test/common/stats/stat_merger_fuzz_test.cc

```diff
@@ -24,26 +24,31 @@ void testDynamicEncoding(absl::string_view data, SymbolTable& symbol_table) {
   // StatMergerDynamicTest.DynamicsWithRealSymbolTable.
   std::string unit_test_encoding;
 
-  for (uint32_t index = 0; index < data.size();) {
+  for (uint32_t index = 0; index < data.size() - 1;) {
+    if (index != 0) {
+      unit_test_encoding += ".";
+    }
+
     // Select component lengths between 1 and 8 bytes inclusive, and ensure it
     // doesn't overrun our buffer.
     //
     // TODO(#10008): We should remove the "1 +" below, so we can get empty
     // segments, which trigger some inconsistent handling as described in that
     // bug.
     uint32_t num_bytes = (1 + data[index]) & 0x7;
-    num_bytes = std::min(static_cast<uint32_t>(data.size() - 1),
-                         num_bytes); // restrict number up to the size of data
+    bool is_symbolic = (data[index] & 0x8) == 0x0;
```

Comment:
nit: move line 45-46 comment about using control-byte to decide if it's symbolic up to this line